### ISSUE TITLE
ngMock $httpBackend should keep a list of the received calls to debug tests

### ISSUE THREAD

**Issue by fabiosussetto** [angular/angular.js#10596]
Hi, I'm not sure if this is already possible, but I haven't found a way yet. In my unit tests I'm using $httpBacked to expect calls to an API. When the expectations fail, it would be useful to see a list of the actual calls, to see you .expectGET('..') failed because your code is calling the wrong url, or is sending the wrong data.

**Comment by Yvem** [comment]
I totally support more details when failing $httpBackend expectations.

Ex. the cryptic "Error: Unflushed requests: 4" : Why can't we access the unflushed requests ? At last their url to give us a lead ?

**Comment by thynctank** [comment]
+1 for accessing unflushed requests

**Comment by turquoiseowl** [comment]
With v1.2.25, the installation of an $exceptionHandler can prevent $httpBackend's  "Unexpected request ..." exception from propagating out to Jasmine. It doesn't seem to affect other $httpBackend-thrown exceptions.

**Comment by senthanal** [comment]
+1 for accessing unflushed requests, stacktrace as well.

**Comment by bostrom** [comment]
Agree. 
```
Error: Unflushed requests: 2
    at Function.$httpBackend.verifyNoOutstandingRequest
```
isn't very informative. I'd like to know what the unflushed requests are.

**Comment by lerzenit** [comment]
+1. 
In my case, I'm testing a factory depending on ngResource using a cache and want to test if the same request is done twice, and I have no way to know wich one is failing.

    server.expectGET('/data').respond([item]);
    server.expectGET('/data').respond([item]);

    var resource = new resource('/data');

    resource.query();
    resource.invalidateCache();
    resource.query();

And this is what I get

    Error: Unsatisfied requests: GET /data
	    at Function.$httpBackend.verifyNoOutstandingExpectation (public/js/angular.js:2290:13)
	    at Function.$httpBackend.flush (public/js/angular.js:2269:18)
	    at Object.<anonymous> (tests/frontend/share/cached-resource.factory.test.js:52:16)


I'm quite new to ngMock and $httpBackend so probably there is a better way to test this. However, I think $httpBackend should provide an error message with more information

**Comment by Narretz** [comment]
We just merged a commit into master / 1.6 that will make the following change:

Before this change:

    Unflushed requests: 1

After this change:

    Unflushed requests: 1
      GET /some


This doesn't cover all cases reported in this issue, only the most requested one: https://github.com/angular/angular.js/issues/10596#issuecomment-227725855

Stack Traces for unexpected requests are tracked here: https://github.com/angular/angular.js/issues/12231
